desc/objectives/_coils.py

```diff
@@ -597,3 +598,151 @@ def compute(self, params, constants=None):
+        coils,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        grid=None,
+        name="coil torsion",
+    ):
+        if target is None and bounds is None:
+            target = 0
+        self._coils = coils
+
+        super().__init__(
+            coils,
+            ["x"],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            grid=grid,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        from desc.coils import CoilSet, _Coil
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+        if self._normalize:
+            self._normalization = self._scales["a"]
+
+        # TODO: repeated code but maybe it's fine
+        flattened_coils = tree_flatten(
+            self._coils,
+            is_leaf=lambda x: isinstance(x, _Coil) and not isinstance(x, CoilSet),
+        )[0]
+        flattened_coils = (
+            [flattened_coils[0]]
+            if not isinstance(self._coils, CoilSet)
+            else flattened_coils
+        )
+        self._dim_f = len(flattened_coils)
+        self._constants["quad_weights"] = 1
+
+    def compute(self, params, constants=None):
+        """Compute coil torsion.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of the coil's degrees of freedom.
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self._constants.
+
+        Returns
+        -------
+        f : array of floats
+            Distance to nearest coil for each coil in the coilset.
+        """
+        data = super().compute(params, constants=constants, basis="xyz")
+        data = tree_flatten(data, is_leaf=lambda x: isinstance(x, dict))[0]
+        # FIXME: what if coilset has NFP or stell symmetry?
+        # we want to compare distances across all coils...
+        min_dists = []
+        for whichCoil1 in range(len(data)):
+            coords1 = data[whichCoil1]["x"]
+            mindist = 1e4
+            for whichCoil2 in range(len(data)):
+                if whichCoil1 == whichCoil2:
+                    continue
+                coords2 = data[whichCoil2]["x"]
+
+                d = jnp.linalg.norm(
+                    coords1[:, None, :] - coords2[None, :, :],
+                    axis=-1,
+                )
+                mindist = jnp.min(jnp.array([jnp.min(d), mindist]))
```

Comment:
I don't think I get why we're taking the minimum here. I thought if we set a low target then the distance would be optimized for that so why do we take the minimum?

Reply:
in the outer loop. I choose a coil, then in the inner loop we loop thru every other coil in the coilset. 

I want one number per coiil which is the min dist from that coil to the other coils in the coilset. the first min (`jnp.min(d)`) gives me the minimum distance from this coil to the current coil I am comparing against. Then the second `jnp.min` call checks if I am closer to this coil than I was to any of the other coils I've already checked previously. If yes, it updates the mindist for this coil to be the smaller value.